sdks/java/core/src/main/java/org/apache/beam/sdk/testing/TestPipeline.java

```diff
@@ -292,8 +298,8 @@ public void evaluate() throws Throwable {
   public PipelineResult run() {
     checkState(
         enforcement.isPresent(),
-        "Attempted to run a pipeline while it's enforcement level was not set. Are you "
-            + "using TestPipeline without a @Rule annotation?");
+        "Is your TestPipeline declaration missing a @Rule annotation? Usage: "
```

Comment:
If that is the case, we might separate the two use-cases into different classes.
One that is implemented as a rule, for the SDK & runners, and the other as a plain non-rule class, for users.

WDYT?